### ISSUE TITLE
Excon::Errors::StubNotFound in test suite

### ISSUE THREAD

**Issue by edwinv** [excon/excon#558]
After upgrading to version 0.47.0, our complete test suite is broken due to an Excon exception. Previous versions don't show this exception. My guess is that commit e5339600ecbef59a9d58ebe407b8addfb6a321f5 has changed something in the Excon Mock behaviour.

Backtrace:

```
excon (0.47.0) lib/excon/middlewares/mock.rb:43:in `request_call'
excon (0.47.0) lib/excon/middlewares/instrumentor.rb:25:in `request_call'
excon (0.47.0) lib/excon/middlewares/base.rb:15:in `request_call'
excon (0.47.0) lib/excon/middlewares/base.rb:15:in `request_call'
excon (0.47.0) lib/excon/middlewares/base.rb:15:in `request_call'
excon (0.47.0) lib/excon/connection.rb:233:in `request'
```

We don't use Excon stubbing/mocking. The request is never begin stubbed, because it is a backing service that is always available in all our environments.

We have downgraded to 0.46.0 for now, this solves all the exceptions.

**Comment by geemus** [comment]
Actually, I suspect this would be a change in the instrumentor behavior introduced here: https://github.com/excon/excon/pull/552

Are you using a custom instrumentor? If so you might need to slightly update the usage to accommodate that change by ensuring it does the right thing wrt blocks. Let me know if that doesn't make sense or you have further questions though.

**Comment by edwinv** [comment]
We use Excon in the most simple variant:

```
connection = Excon.new(url)
params = { ... }
response = connection.request(params)
```

We don't use any other Gems that have a dependency on Excon, so I guess no other gem is adding instrumentation for Excon (although this is hard to check...).

**Comment by geemus** [comment]
Do you use VCR? It integrates with this stuff, so might be a culprit anyway.

**Comment by pironim** [comment]
I have similar problem in development .. Excon returns this error. but when you use console it works fine.. is there any way to avoid mocking by default or have explicit config to enable mocking?

**Comment by edwinv** [comment]
> Do you use VCR? It integrates with this stuff, so might be a culprit anyway.

Yes, we use VCR. I don't have the opportunity to test without VCR at this time.

**Comment by pironim** [comment]
@edwinv I also use VCR and tests ignore VCR and perform regular call with Excon.post(...mock: false)
But the problem is different for some reason I have mock: true in development mode(Rails) and can't figure out how to trace it. 
<pre><code>
-> Excon.defaults
{:chunk_size=>1048576,
 :ciphers=>"HIGH:!SSLv2:!aNULL:!eNULL:!3DES",
 :connect_timeout=>60,
 :debug_request=>false,
 :debug_response=>false,
 :headers=>{"User-Agent"=>"excon/0.48.0"},
 :idempotent=>false,
 :instrumentor_name=>"excon",
 :middlewares=>
  [Excon::Middleware::ResponseParser,
   Excon::Middleware::Expects,
   Excon::Middleware::Idempotent,
   Excon::Middleware::Instrumentor,
   Excon::Middleware::Mock],
 :mock=>true,
 :nonblock=>true,
 :omit_default_port=>false,
 :persistent=>false
}

**Comment by geemus** [comment]
I'll have to try and dig a bit deeper, sorry for the trouble here. 0.46.0 was released prior to this behavior change, so in the short-term you could downgrade as a workaround.

@edwinv Is VCR also included in development mode? It might be related, though I'm not certain of that.

**Comment by geemus** [comment]
I just made a pull that allows you to turn this behavior back off, could you check it out and see if that helps? https://github.com/excon/excon/pull/563 Thanks!

**Comment by geemus** [comment]
closed by #563
